### ISSUE TITLE
Run demos throw an error,I can not find the reason.

### ISSUE THREAD

**Issue by TJ666** [PAIR-code/deeplearnjs#50]
When I run the demo, terminal throw an error, my node verson is 0.11.0,it is lower?Please help me. I come form China, my English is poor, sorry~

$ ./scripts/watch-demo demos/nn-art/nn-art.ts
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules\.bin\watchify ENOENT
    at exports._errnoException (util.js:1018:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:367:16)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3

**Comment by nsthorat** [comment]
Have you run 'npm run prep'?

**Comment by TJ666** [comment]
Yes,I did.There are some Warn,but no error.Dependence looks like no problems.

v_yangtianjiao@6FD27AF4094B0D2 MINGW32 /d/learn/learn-js/deeplearnjs (master)
$ cnpm run prep

> deeplearn@0.1.0 prep D:\learn\learn-js\deeplearnjs
> npm install && bower install && mkdirp dist

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current
: {"os":"win32","arch":"x64"})
npm WARN deeplearn@0.1.0 No license field.

**Comment by dsmilkov** [comment]
Looks like you are using windows based on your path in logs `D:\learn\learn-js\deeplearnjs`

If that case, can you try running
$ ./scripts/watch-demo "demos\nn-art\nn-art.ts" (note the \ instead of /)

If that doesn't work try:
$ ./scripts/watch-demo "demos\\nn-art\\nn-art.ts" (note the \\ instead of /)

And please let us know, so we can update your scripts and docs.

**Comment by TJ666** [comment]
It does't work. My colleague'win7 has the same problem. It is maybe beacuse win7 env. Is my understanding is wrong?  Have you try it on win7?

```
v_yangtianjiao@6FD27AF4094B0D2 MINGW32 /d/learn/learn-js/deeplearnjs (master)
$ scripts\watch-demo "demos\nn-art\nn-art.ts"
bash: scriptswatch-demo: command not found
```

```
v_yangtianjiao@6FD27AF4094B0D2 MINGW32 /d/learn/learn-js/deeplearnjs (master)
$ scripts\watch-demo "demos\\nn-art\\nn-art.ts"
bash: scriptswatch-demo: command not found
```

```
v_yangtianjiao@6FD27AF4094B0D2 MINGW32 /d/learn/learn-js/deeplearnjs (master)
$ ./scripts/watch-demo "demos\\nn-art\\nn-art.ts"
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules\.bin\watchify ENOENT
    at exports._errnoException (util.js:1018:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:367:16)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
    at Module.runMain (module.js:606:11)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3

```

**Comment by JainVikas** [comment]
I am on win10 and facing same problem.

`vikas.e.jain@BDC11-L-72000H0 MINGW64 ~/desktop/deeplearnjs (master)
$ .\\scripts\\watch-demo demos\\complementary-color-prediction\\complementary-color-prediction.ts`
events.js:182
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules\.bin\watchify ENOENT
    at _errnoException (util.js:1041:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:192:19)
    at onErrorNT (internal/child_process.js:374:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
    at Function.Module.runMain (module.js:611:11)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:598:3

`npm -v`
5.3.0
`node -v`
v8.4.0

` npm install && bower install && mkdirp dist`

npm WARN deeplearn@0.1.0 No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

**Comment by Hiwap** [comment]
@JainVikas You can install bash for windows 10 and getting working 💯

**Comment by JainVikas** [comment]
@Hiwap , i was using Git bash for running this, will that be okay or should i use this https://msdn.microsoft.com/en-us/commandline/wsl/install_guide ?

**Comment by MichaelCharrier** [comment]
I believe I found the problem, and an alright workaround for Windows users.

The problem stems from the scripts using nodejs's ChildProcess.spawn() command to execute other scripts in separate processes. There are known problems with using the spawn() command in windows (see here: [Error: spawn ENOENT” on node.js?](https://stackoverflow.com/questions/27688804/how-do-i-debug-error-spawn-enoent-on-node-js)).

The solution I found was to use nodejs's ChildProcess.exec() instead. It buffers output from the scripts until completion and using exec is considered less safe, but they still run.

There was also a problem with paths having mixed slashes, so I went ahead and explicitly made them all posix style (using "/").

Attached are the windows versions of the build-demo and watch-demo scripts. Look them over to make sure they make sense before using them (remove the .txt and run using bash).

Let me know if you have any more troubles.

[windows-build-demo.txt](https://github.com/PAIR-code/deeplearnjs/files/1240728/windows-build-demo.txt)
[windows-watch-demo.txt](https://github.com/PAIR-code/deeplearnjs/files/1240729/windows-watch-demo.txt)
